simpleclient_logback/src/test/java/io/prometheus/client/logback/InstrumentedAppenderTest.java

```diff
@@ -12,65 +12,88 @@
-    when(event.getLevel()).thenReturn(Level.TRACE);
+    @Test
+    public void metersTraceEvents() throws Exception {
+      when(this.event.getLevel()).thenReturn(Level.TRACE);
 
-    appender.doAppend(event);
+      this.appender.doAppend(event);
 
-    assertEquals(1, getLogLevelCount("trace"));
-  }
+      assertEquals(1, this.getLogLevelCount("trace"));
+    }
 
-  @Test
-  public void metersDebugEvents() throws Exception {
-    when(event.getLevel()).thenReturn(Level.DEBUG);
+    @Test
+    public void metersDebugEvents() throws Exception {
+      when(this.event.getLevel()).thenReturn(Level.DEBUG);
 
-    appender.doAppend(event);
+      this.appender.doAppend(event);
 
-    assertEquals(1, getLogLevelCount("debug"));
-  }
+      assertEquals(1, this.getLogLevelCount("debug"));
+    }
 
-  @Test
-  public void metersInfoEvents() throws Exception {
-    when(event.getLevel()).thenReturn(Level.INFO);
+    @Test
+    public void metersInfoEvents() throws Exception {
+      when(this.event.getLevel()).thenReturn(Level.INFO);
 
-    appender.doAppend(event);
+      this.appender.doAppend(event);
 
-    assertEquals(1, getLogLevelCount("info"));
-  }
+      assertEquals(1, this.getLogLevelCount("info"));
+    }
+
+    @Test
+    public void metersWarnEvents() throws Exception {
+      when(this.event.getLevel()).thenReturn(Level.WARN);
 
-  @Test
-  public void metersWarnEvents() throws Exception {
-    when(event.getLevel()).thenReturn(Level.WARN);
+      this.appender.doAppend(event);
 
-    appender.doAppend(event);
+      assertEquals(1, this.getLogLevelCount("warn"));
+    }
 
-    assertEquals(1, getLogLevelCount("warn"));
+    @Test
+    public void metersErrorEvents() throws Exception {
+      when(this.event.getLevel()).thenReturn(Level.ERROR);
+
+      this.appender.doAppend(event);
+
+      assertEquals(1, this.getLogLevelCount("error"));
+    }
+
+    private int getLogLevelCount(String level) {
+      return this.registry.getSampleValue(COUNTER_NAME,
+              new String[]{"level"}, new String[]{level}).intValue();
+    }
   }
 
-  @Test
-  public void metersErrorEvents() throws Exception {
-    when(event.getLevel()).thenReturn(Level.ERROR);
+  public static class DefaultTest extends Base {
+    @Before
+    public void setUp() throws Exception {
+      this.registry = CollectorRegistry.defaultRegistry;
 
-    appender.doAppend(event);
+      this.appender = new InstrumentedAppender();
 
-    assertEquals(1, getLogLevelCount("error"));
+      super.setUp();
+    }
   }
 
-  private int getLogLevelCount(String level) {
-    return CollectorRegistry.defaultRegistry.getSampleValue(COUNTER_NAME, 
-            new String[]{"level"}, new String[]{level}).intValue();
+  public static class InstanceTest extends Base {
+    @Before
+    public void setUp() throws Exception {
+      this.registry = new CollectorRegistry();
+
+      this.appender = new InstrumentedAppender(this.registry);
+
+      super.setUp();
+    }
```

Comment:
Thanks for the PR, the changes to the production code are looking good.

However, in the test code I think that the inner classes `DefaultTest` and `InstanceTest` are not executed in the Maven build. If I change an `assertEquals()` to make a test fail, the Maven build is still green. In my opinion there's no need for the `DefaultTest`, you could make `InstrumentedAppenderTest` use a custom registry and remove the inner classes.

Reply:
Thanks for the review. Good catch.

I reverted the change and did something more simple/less fancy. I added two appenders (one default and one for a custom registry) into the test and then just appended to both of them and checked both of them. That should work.

Happy to revert even this if you would prefer the simplicity of testing only with the custom registry. Just wanted to push it up as an alternative option.

Tested locally by changing an `assertEquals(...)` this time.